src/components/orders/OrdersUserDetailsTable/index.tsx

```diff
@@ -0,0 +1,121 @@
+import React from 'react'
+import styled from 'styled-components'
+import BigNumber from 'bignumber.js'
+
+import { calculatePrice, TokenErc20, formatSmart } from '@gnosis.pm/dex-js'
+import { Order } from 'api/operator'
+
+import { DateDisplay } from 'components/orders/DateDisplay'
+import { RowWithCopyButton } from 'components/orders/RowWithCopyButton'
+import { formatSmartMaxPrecision } from 'utils'
+import { StatusLabel } from '../StatusLabel'
+import { HelpTooltip } from 'components/Tooltip'
+import {
+  HIGH_PRECISION_DECIMALS,
+  HIGH_PRECISION_SMALL_LIMIT,
+  NO_ADJUSTMENT_NEEDED_PRECISION,
+} from 'apps/explorer/const'
+import TradeOrderType from '../TradeOrderType'
+import StyledUserDetailsTable, { StyledUserDetailsTableProps } from './styled'
+
+const Wrapper = styled(StyledUserDetailsTable)`
+  > thead > tr,
+  > tbody > tr {
+    grid-template-columns: 12rem 23rem repeat(2, 13rem) 16rem 10rem 1fr;
+  }
+`
+function isTokenErc20(token: TokenErc20 | null | undefined): token is TokenErc20 {
+  return (token as TokenErc20).address !== undefined
+}
+
+const formattedAmount = (erc20: TokenErc20 | null | undefined, amount: BigNumber): string => {
+  if (!isTokenErc20(erc20)) return '-'
+
+  return erc20.decimals ? formatSmartMaxPrecision(amount, erc20) : amount.toString(10)
+}
+
+const getLimitPrice = (order: Order): string => {
```

Comment:
@alfetopito I have used as reference the logic in `OrderPriceDisplay` [component](https://github.com/gnosis/gp-ui/blob/1044-orders-userdetails-table/src/components/orders/OrderPriceDisplay/index.tsx#L52), could it be taken out to reuse here?

Reply:
Yes, that's a good idea.

Actually, @stackseeder is working on exactly the same thing, you guys are overlapping.

I commented on his PR https://github.com/gnosis/gp-ui/pull/569#discussion_r687239936

So my suggestion is either:
1. Use the function as you have it, extracting and updating OrdersPriceDisplay
2. Use the utils functions implemented and create another utils for formatting the price

I wrote those helpers (https://github.com/gnosis/gp-ui/blob/36eae4570aac573e71d909042d3d4b18adfbf5aa/src/utils/operator.ts#L172 and https://github.com/gnosis/gp-ui/blob/36eae4570aac573e71d909042d3d4b18adfbf5aa/src/utils/operator.ts#L195) but I cannot remember now why I did not use them.

Reply:
Also, just noticed that your code does not account for the price being inverted, which should be possible to do for the whole table at once.